From f0557941510ab97187a1b27f79b8e89b03b13b9d Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Thu, 18 Nov 2021 20:39:59 +0200
Subject: [PATCH] [SW-36216] gaudi2: sec: change bootstrap check

SECURITY_BYPASS boot strap pin in gaudi2 is used for
enabling/disabling arcboot flow (connected to GPIO_47).
Hence strap check to be confirmed with system team if
additional indicator like otp is needed to make the solution
robust like gaudi.

Change-Id: Ifd0cc4cde0a69186192987e7d7daa383e14378ea
---

diff --git a/arch/gaudi2/arm/common.c b/arch/gaudi2/arm/common.c
index 3d89e9e..212dda5 100644
--- a/arch/gaudi2/arm/common.c
+++ b/arch/gaudi2/arm/common.c
@@ -409,23 +409,17 @@
 
 #define EFUSE_SECURITY_ENABLE_BIT		BIT(15)
 #define EFUSE_SECURITY_WORD_OFFSET		0x3A
-#define BOOT_STRAP_SECURITY_ENABLE_BIT		BIT(0)
 
-/*
- * The routine shall return true in case of security enable in strap.
- * Otherwise, shall return false (The polarity of strap is inverted:
- * meaning '0' is for security enable).
- */
-static bool hl_is_secuirty_strap_enabled(void)
+static bool hl_is_security_strap_enabled(void)
 {
-	u32 boot_strap;
-
-	boot_strap = regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS_H);
-
-	if (boot_strap & BOOT_STRAP_SECURITY_ENABLE_BIT)
-		return false;
-
-	return true;
+	/*
+	 * TODO: SECURITY_BYPASS boot strap pin in gaudi2
+	 * is used for enabling/disabling arcboot flow (connected
+	 * to GPIO_47). Hence strap check to be confirmed with
+	 * system team if additional indicator like otp is needed
+	 * to make the solution robust like gaudi.
+	 */
+	return false;
 }
 
 bool preboot_security_is_enabled(void)
@@ -433,7 +427,7 @@
 	u32 efuse_word;
 
 	/* If security is enabled via boot strap pins */
-	if (hl_is_secuirty_strap_enabled())
+	if (hl_is_security_strap_enabled())
 		return true;
 
 	/* If eFuse read failed, assume an attacker sabotage, security in on */
